### DISCUSSION TITLE
How to implement customized initialization of population?

### DISCUSSION THREAD

**Discussion by csjtx1021** [MilesCranmer/PySR#491] [Q&A] [answered]
Hi all,

I am implementing a custom project and have found the library very helpful.  How can I set up a customized initialization population to trigger symbol regression? Thanks in advance.

Jay

**Comment by MilesCranmer** [comment, marked as answer]
Hi Jay,

The current code for initializing the population of expressions is here:

https://github.com/MilesCranmer/SymbolicRegression.jl/blob/f38ea0df4c362cd52d0cc2034c358c0b11474697/src/Population.jl#L37-L63

Right now it calls `gen_random_tree(nlength, options, nfeatures, T)` which is basically appending `nlength` random nodes to a root node. The code for that method is here: https://github.com/MilesCranmer/SymbolicRegression.jl/blob/f38ea0df4c362cd52d0cc2034c358c0b11474697/src/MutationFunctions.jl#L228-L238.

If you're using PySR (?) you can see https://astroautomata.com/PySR/backend/ for customizing the backend and implementing custom behavior. (But it's probably easier to start directly with SymbolicRegression.jl so you can use Revise.jl for quickly changing things).

Cheers,
Miles

**Reply by csjtx1021** [reply]
It seems to have a high success rate to solve my problem. Appreciate it.

Jay
